bin/genome-env-next

```diff
@@ -0,0 +1,432 @@
+            echo 'Press <Ctrl>-D or type `logout` to return to normal environment.'
+            echo ''
+            if test -z "$GE_ENV_NAME"
+            then
+                GE_ENV_NAME="genome-env"
+            fi
+            unset PROMPT_COMMAND
+            export PS1="(${GE_ENV_NAME}) \w $ "
+            exec bash --norc --noprofile
+        else
+            echo 'Only `bash` shells are supported at this time.'
+            exit 128
+        fi
+    fi
+}
+
+function err {
+    echo "$THIS: $@" >&2
+}
+
+function parse_opts {
+    local opts
+    while getopts :hMSs:Uu:vWw: opts "$@"
+    do
+        case $opts in
+            h)
+                help
+                exit 0
+                ;;
+            M)
+                DISABLE_MIGRATIONS="TRUE"
+                ;;
+            S)
+                DISABLE_DB="TRUE"
+                ;;
+            s)
+                SNAPSHOT_NAME="$OPTARG"
+                ;;
+            U)
+                DISABLE_UR="TRUE"
+                ;;
+            u)
+                UR_REPO="$OPTARG"
+                ;;
+            v)
+                VERBOSE="TRUE"
+                ;;
+            W)
+                DISABLE_WF="TRUE"
+                ;;
+            w)
+                WF_REPO="$OPTARG"
+                ;;
+            :)
+                err "option -$OPTARG requires an argument"
+                exit 1
+                ;;
+            \?)
+                err "illegal option -- $OPTARG"
+                usage
+                exit 1
+                ;;
+        esac
+    done
+
+    validate_opts
+    set_defaults
+}
+
+function validate_opts {
+    if test -n "$DISABLE_DB" -a -n "$DISABLE_MIGRATIONS"
+    then
+        err "cannot use -S and -M options together"
+        exit 2
+    fi
+
+    if test -n "$DISABLE_DB" -a -n "$SNAPSHOT_NAME"
+    then
+        err "cannot use -S and -s options together"
+        exit 2
+    fi
+
+    if test -n "$DISABLE_UR" -a -n "$UR_REPO"
+    then
+        err "cannot use -U and -u options together"
+        exit 2
+    fi
+
+    if test -n "$DISABLE_WF" -a -n "$WF_REPO"
+    then
+        err "cannot use -W and -w options together"
+        exit 2
+    fi
+
+    if test -n "$SNAPSHOT_NAME"
+    then
+        if ! test-db template list | cut -f 1 | grep -q "^${SNAPSHOT_NAME}$"
+        then
+            err "invalid snapshot name -- $SNAPSHOT_NAME"
+            exit 2
+        fi
+    fi
+
+    if test -n "$UR_REPO"
+    then
+        assert_dir "-u" "$UR_REPO"
+    fi
+
+    if test -n "$WF_REPO"
+    then
+        assert_dir "-w" "$WF_REPO"
+    fi
+}
+
+function assert_dir {
+    if ! test -d "$2"
+    then
+        err "$1 value must be a directory -- $2"
+        exit 2
+    fi
+}
+
+function set_defaults {
+    local TOP="$(git rev-parse --show-toplevel 2> /dev/null)"
+    local SUBMODULES
+
+    SUBMODULES=(gms-config configuration-manager)
+
+    if test -z "$SNAPSHOT_NAME"
+    then
+        SNAPSHOT_NAME="82f3e4d"
```

Comment:
It's a template name used to create the ephemeral database that happens to be names after a commit hash.  It's definitely data.  I haven't decided where to store it yet.  It may need to be tightly coupled to the source revision.